remote/bitbucketserver/bitbucketserver.go

```diff
@@ -152,20 +161,28 @@ func (c *client) Repo(u *model.User, owner, name string) (*model.Repo, error) {
 	defer response.Body.Close()
 	contents, err := ioutil.ReadAll(response.Body)
 	bsRepo := BSRepo{}
-	json.Unmarshal(contents, &bsRepo)
-
+	err = json.Unmarshal(contents, &bsRepo)
+	if err !=nil {
+		return nil, err
+	}
 	repo := &model.Repo{
 		Name:      bsRepo.Slug,
 		Owner:     bsRepo.Project.Key,
 		Branch:    "master",
 		Kind:      model.RepoGit,
-		IsPrivate: !bsRepo.Project.Public, // TODO(josmo) verify this is corrrect
+		IsPrivate: true, // TODO(josmo) possibly set this as a setting - must always be private to use netrc
 		FullName:  fmt.Sprintf("%s/%s", bsRepo.Project.Key, bsRepo.Slug),
 	}
 
 	for _, item := range bsRepo.Links.Clone {
 		if item.Name == "http" {
-			repo.Clone = item.Href
+			//TODO sdhould find a clean way to do this
+			//We are removing the username out of the link to allow for Netrc to work
+			splitUrl := strings.SplitAfterN(item.Href,"@",2)
```

Comment:
That was my first thought.  I just haven't used the golang libs enough but I'll take a look tomorrow at switching it that way.  It would be cleaner.